compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -205,6 +206,43 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
         _ => elf::ELFOSABI_NONE,
     };
     let abi_version = 0;
+    // check bti protection
+    let mut check_cfprotection = false;
+    let mut pr_type: u32 = 0;
+    if architecture == Architecture::X86_64 && let rustc_session::config::CFProtection::Branch | rustc_session::config::CFProtection::Full = sess.opts.unstable_opts.cf_protection {
+      check_cfprotection =true;
```

Comment:
How did rustfmt not flag this missing space after the `=`?

Reply:
Probably because it's in a let chain, which are ignored by rustfmt (in this repo, I don't know about it in general).

Rewriting this to something like `matches!(sess.opts.unstable_opts.cf_protection, rustc_session::config::CFProtection::Branch | rustc_session::config::CFProtection::Full)`  would likely be better and rustfmt-able.

Reply:
changed the code and it became rustfmt-able. Thanks!